lib/muser.h

```diff
@@ -377,10 +377,11 @@ enum {
 /**
  * Setup device IRQ counts.
  * @lm_ctx: the libmuser context
+ * @irq_idx: IRQ idx (LM_DEV_INTX_IRQ_IDX ... LM_DEV_REQ_IRQ_INDEX)
  * @irq_count: array of IRQ count value
  */
-int lm_setup_device_irq_counts(lm_ctx_t *lm_ctx,
-                               uint32_t irq_count[LM_DEV_NUM_IRQS]);
+int lm_setup_device_irq_counts(lm_ctx_t *lm_ctx, int irq_idx,
```

Comment:
Why do we need this? Can't the use simply pass `[1, 0, 0, 0, 0]`? Otherwise we'll have to call this function as many times as IRQs we want to set.

Reply:
I had it previously. Later I realised that just passing array is not sufficient, we will also need the size of array to be passed as additional argument.
I thought its easier for user to pass irq_count individually, instead creating an array after looking the values of LM_DEV_INTX_IRQ_IDX .. LM_DEV_REQ_IRQ_INDEX

Reply:
I prefer individually too
I wonder if we should call them "index" here - that was a horribly confusing part of vfio we don't need to replicate here. We can have:

int lm_setup_nr_irqs(lm_ctx_t *lm_ctx, enum lm_irq_type type, size_t count);

?

Reply:
Why do we need to pass the size of the array? There are only 5 IRQ types.

Reply:
We could easily get new types, let's not bake this in (and equally an array is not really great for this). I think separate calls are *much* easier to read and write.